### PR TITLE
XML documentation comments for StringValues.

### PR DESCRIPTION
**Pull request by benbhall**
- Add XML comment to clarify behaviour casting to string as per issue.
- Add XML across StringValues members.
- Minor tidy with some expression bodies.

Addresses #2662

### CODE REVIEW

**Comment by benbhall** [comment]
Ah okay. I'm probably distracted by the issue being addressed around undefined behaviour, so I've tried to take the surprise out of interface where there could be confusion.

Is it a worthwhile piece? i.e. shall I have another bash at it? Any further tips welcome. At least I got it right for `Count` :)

**Comment by anurse** [comment]
> Is it a worthwhile piece?

I think docs for this are definitely worthwhile. It's the type used as the "value" for HTTP headers, and some doc clarity would be very valuable.

**Comment by benbhall** [comment]
Okay second attempt in @anurse.

I've tried to adhere to your advice re documenting **how to use** the members and just used `<remark>` to supplement:

- `Empty`
- The `implicit operator string (StringValues values)` I believe #2662 concerned
- `ToArray()` re violation principle of least astonishment raised in #1814 

Hoping to pick up the latter when I've finished with these docs if that's okay.

File: src/Primitives/src/StringValues.cs [hunk 1]
```diff
@@ -112,6 +159,10 @@ private static string OutOfBounds()
             return Array.Empty<string>()[0]; // throws
         }
 
+        /// <summary>
+        /// Converts the value of the current <see cref="StringValues"/> object to its equivalent string representation.
+        /// </summary>
+        /// <returns>A string representation of the value of the current <see cref="StringValues"/> object.</returns>
```
Comment (Tratcher) [inline src/Primitives/src/StringValues.cs]: With multiple entries being comma separated.

File: src/Primitives/src/StringValues.cs [hunk 2]
```diff
@@ -240,6 +299,11 @@ private string[] GetArrayValue()
             }
         }
 
+        /// <summary>
+        /// Returns the zero-based index of the first occurrence of a item in the <see cref="StringValues" />.
```
Comment (Tratcher) [inline src/Primitives/src/StringValues.cs]:
```suggestion
        /// Returns the zero-based index of the first occurrence of an item in the <see cref="StringValues" />.
```

File: src/Primitives/src/StringValues.cs [hunk 3]
```diff
@@ -433,6 +520,12 @@ public static StringValues Concat(string value, in StringValues values)
             return new StringValues(combined);
         }
 
+        /// <summary>
+        /// Determines whether two specified <see cref="StringValues"/> objects have the same values.
```
Comment (Tratcher) [inline src/Primitives/src/StringValues.cs]:
```suggestion
        /// Determines whether two specified <see cref="StringValues"/> objects have the same values in the same order. 
```

File: src/Primitives/src/StringValues.cs [hunk 4]
```diff
@@ -15,40 +15,76 @@ namespace Microsoft.Extensions.Primitives
...
         public static implicit operator StringValues(string[] values)
         {
             return new StringValues(values);
         }
 
+        /// <summary>
+        /// Defines an implicit conversion of a given <see cref="StringValues"/> to a string.
+        /// </summary>
+        /// <remarks>
+        /// Returns <c>null</c> where <see cref="StringValues"/> has been initialized from an empty string array or is <see cref="StringValues.Empty"/>.
```
Comment (Tratcher) [inline src/Primitives/src/StringValues.cs]: with multiple values joined as a comma separated string.

File: src/Primitives/src/StringValues.cs [hunk 5]
```diff
@@ -216,6 +267,14 @@ static string GetJoinedStringValueFromArray(string[] values)
             }
         }
 
+        /// <summary>
+        /// Creates a string array from the current <see cref="StringValues"/> object.
+        /// </summary>
+        /// <returns>A string array represented by this instance.</returns>
+        /// <remarks>
+        /// <para>If the <see cref="StringValues"/> contains a string internally, it is copied to a new array.</para>
```
Comment (Tratcher) [inline src/Primitives/src/StringValues.cs]:
```suggestion
        /// <para>If the <see cref="StringValues"/> contains a single string internally, it is copied to a new array.</para>
```

File: src/Primitives/src/StringValues.cs (hunk 3 above)
Comment (benbhall) [reply src/Primitives/src/StringValues.cs]: Good spot.

File: src/Primitives/src/StringValues.cs (hunk 1 above)
Comment (benbhall) [reply src/Primitives/src/StringValues.cs]: Yep good idea will push something up now.

**Comment by Tratcher** [comment]
Thanks

**Pull request merged**